en-us/api/Microsoft_Toolkit_Uwp_UI_FileCache.md

```diff
@@ -0,0 +1,101 @@
+---
+permalink: /en-US/api/Microsoft_Toolkit_Uwp_UI_FileCache.htm
+title: Microsoft.Toolkit.Uwp.UI.FileCache API 
+description: API page for Microsoft.Toolkit.Uwp.UI.FileCache
+keywords: windows, app, toolkit, UWP, API
+layout: api
+search.product: eADQiWindows 10XVcnh
+---
+
+
+# FileCache class
+
+Represents a cache of StorageFile objects that can be added to and removed as desired.
+The cache also provides in-memory storage of files to reduce reading from disc. The size of this in-memory storage is controlled by setting the ***MaxMemoryCacheCount*** property.
+The in-memory storage follows FIFO (first in first out) and the caller can pass optional parameter when calling ***PreCacheAsync*** to inform cache to use in-memory storage.
+
+## Members
+
+The **FileCache** class has the following types of members:
+
+### Methods
+
+#### InitializeAsync(Windows.Storage.StorageFolder folder, System.String folderName)
+
+this method is used supply a folder within which cache specific folder is created based on name provided.
+
+##### Parameters
+######| Name | Description | Type |
+
+| folder | folder where cache specific folder is created | [StorageFolder](https://msdn.microsoft.com/en-us/library/windows/apps/windows.storage.storagefolder.aspx) |
+
+| folderName | the name given to cache specific folder | [String](https://msdn.microsoft.com/en-us/library/windows/apps/system.string.aspx) |
+
+| returns | [Task](https://msdn.microsoft.com/en-us/library/windows/apps/system.threading.tasks.task.aspx) |
+
+
+#### ClearAsync()
+
+this method clears the entire cache and any files held within the in-memory storage.
+
+##### Parameters
+######| Name | Description | Type |
+
+| returns | [Task](https://msdn.microsoft.com/en-us/library/windows/apps/system.threading.tasks.task.aspx) |
+
+
+#### ClearAsync(System.TimeSpan duration)
+
+this method delete expired files from the cache and from within the in-memory cache.
+
+##### Parameters
+###### | Name | Description | Type |
+
+| duration | this parameter is used to determine where file within cache has expired and should be delete or not. | [TimeSpan](https://msdn.microsoft.com/en-us/library/windows/apps/system.timespan.aspx) |
```

Comment:
"determine whether the file within the cache has expired and should be deleted or not"